weave/trace/object_record.py

```diff
@@ -64,8 +65,65 @@ def dataclass_object_record(obj: Any) -> ObjectRecord:
     if not dataclasses.is_dataclass(obj):
         raise ValueError(f"{obj} is not a dataclass")
     attrs = dataclass_asdict_one_level(obj)
-    for k, v in inspect.getmembers(obj, lambda x: isinstance(x, Op)):
+    for k, v in getmembers(obj, lambda x: isinstance(x, Op)):
         attrs[k] = v
     attrs["_class_name"] = obj.__class__.__name__
     attrs["_bases"] = class_all_bases_names(obj.__class__)
     return ObjectRecord(attrs)
+
+
+# This is an exact copy of the getmembers function from the inspect module
+# with the addition of handling exceptions when calling getattr
+def getmembers(object: Any, predicate: Any = None) -> list[tuple[str, Any]]:
+    """Return all members of an object as (name, value) pairs sorted by name.
+    Optionally, only return members that satisfy a given predicate."""
+    if isclass(object):
+        mro = (object,) + getmro(object)
+    else:
+        mro = ()
+    results = []
+    processed = set()
+    names = dir(object)
+    # :dd any DynamicClassAttributes to the list of names if object is a class;
+    # this may result in duplicate entries if, for example, a virtual
+    # attribute with the same name as a DynamicClassAttribute exists
+    try:
+        for base in object.__bases__:
+            for k, v in base.__dict__.items():
+                if isinstance(v, types.DynamicClassAttribute):
+                    names.append(k)
+    except AttributeError:
+        pass
+    for key in names:
+        # First try to get the value via getattr.  Some descriptors don't
+        # like calling their __get__ (see bug #1785), so fall back to
+        # looking in the __dict__.
+        try:
+            value = getattr(object, key)
+            # handle the duplicate key
+            if key in processed:
+                raise AttributeError
+        except AttributeError:
+            for base in mro:
+                if key in base.__dict__:
+                    value = base.__dict__[key]
+                    break
+            else:
+                # could be a (currently) missing slot member, or a buggy
+                # __dir__; discard and move on
+                continue
+
+        # This is where the modified code begins
+        except Exception as e:
+            # This is for the one off case where the xmltodict library is not installed in Instructor's environment
```

Comment:
It might be cleaner separation of concerns to have the caller optionally pass in an `on_error` handler that accepts the exception. If present, call during the catch. It can print/raise/etc... Furthermore, i would not print this message. It is not relevant to the user in these cases.